nncf/pruning/export_helpers.py

```diff
@@ -169,6 +170,68 @@ def output_prune(self, model: NNCFNetwork, nx_node, graph: NNCFGraph, nx_graph:
                          ' {}.'.format(nx_node['key'], old_num_clannels, node_module.out_channels))
 
 
+@PRUNING_OPERATOR_METATYPES.register('transpose_convolution')
+class TransposeConvolution(DefaultMetaOp):
+    subtypes = [ConvTranspose2dMetatype, ConvTranspose3dMetatype]
+
+    def mask_propagation(self, model: NNCFNetwork, nx_node, graph: NNCFGraph, nx_graph: nx.DiGraph):
+        output_mask = None
+        accept_pruned_input = True
+        input_masks = get_input_masks(nx_node, nx_graph)
+
+        nncf_node = graph._nx_node_to_nncf_node(nx_node)
+        node_module = model.get_module_by_scope(nncf_node.op_exec_context.scope_in_model)
+
+        if node_module.pre_ops:
+            output_mask = node_module.pre_ops['0'].op.binary_filter_pruning_mask
+
+        nx_node['input_masks'] = input_masks
+        nx_node['output_mask'] = output_mask
+        nx_node['accept_pruned_input'] = accept_pruned_input
+
+    def input_prune(self, model: NNCFNetwork, nx_node, graph: NNCFGraph, nx_graph: nx.DiGraph):
+        input_mask = nx_node['input_masks'][0]
+        if input_mask is None:
+            return
+        bool_mask = torch.tensor(input_mask, dtype=torch.bool)
+
+        nncf_node = graph._nx_node_to_nncf_node(nx_node)
+        node_module = model.get_module_by_scope(nncf_node.op_exec_context.scope_in_model)
+        old_num_clannels = int(node_module.weight.size(0))
+
+        node_module.in_channels = int(torch.sum(bool_mask))
+        node_module.weight = torch.nn.Parameter(node_module.weight[bool_mask])
+
+        nncf_logger.info('Pruned ConvTranspose {} by input mask. Old input filters number: {}, new filters number:'
+                         ' {}.'.format(nx_node['key'], old_num_clannels, node_module.in_channels))
+
+    def output_prune(self, model: NNCFNetwork, nx_node, graph: NNCFGraph, nx_graph: nx.DiGraph):
+        output_mask = nx_node['output_mask']
+        if output_mask is None:
+            return
+
+        bool_mask = torch.tensor(output_mask, dtype=torch.bool)
+        new_num_channels = int(torch.sum(bool_mask))
+
+        nncf_node = graph._nx_node_to_nncf_node(nx_node)
+        node_module = model.get_module_by_scope(nncf_node.op_exec_context.scope_in_model)
+        old_num_clannels = int(node_module.weight.size(0))
```

Comment:
Old num channels are the number of output channels, for ConvTrabspose it should be .size(1)